app/lib/actions/candlepin/listen_on_candlepin_events.rb

```diff
@@ -180,10 +180,11 @@ def finish_service
       def act_on_event(event)
         begin
           output[:connection] = "Connected"
-          on_event(event)
+          Actions::Candlepin::ImportPoolHandler.new(Rails.logger).handle(event)
+          output[:last_message] = "#{event.message_id} - #{event.subject}"
+          output[:messages] = event.message_id
         rescue => e
-          error!(e)
-          raise e
+          output[:last_event_error] = e.message
```

Comment:
I suppose this implies if there are multiple errors coming through, you will only know the last?